### PR TITLE
DOCSP-34994: socketsettings parameter types fix

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-34994>

**this PR also includes content for the removal described in [DOCSP-34104](https://jira.mongodb.org/browse/DOCSP-34104)**

Staging:
- [whats new](https://preview-mongodbrustagir.gatsbyjs.io/java/DOCSP-34994-clustersettings-types-fix/whats-new/)
- [upgrade guide](https://preview-mongodbrustagir.gatsbyjs.io/java/DOCSP-34994-clustersettings-types-fix/upgrade/)
- [example](https://preview-mongodbrustagir.gatsbyjs.io/java/DOCSP-34994-clustersettings-types-fix/fundamentals/connection/mongoclientsettings/#example-5)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade.txt [hunk 1]
```diff
@@ -52,6 +56,27 @@ changes between the current and upgrade versions. For example, if you
...
+.. _java-breaking-changes-v5.0:
+
+Version 5.0 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- The driver removes the following record annotations:
+  
+  - ``BsonId``
+  - ``BsonProperty``
+  - ``BsonRepresentation``
```
Comment (rachel-mack) [inline source/upgrade.txt:68]: Was this meant to be included in this PR?

File: source/upgrade.txt [hunk 2]
```diff
@@ -52,6 +56,27 @@ changes between the current and upgrade versions. For example, if you
...
+
+- The first parameter for each of the following ``SocketSettings`` builder
+  methods is updated from type ``int`` to ``long``:
```
Comment (rachel-mack) [inline source/upgrade.txt:71]: Suggestion: Add more context
```suggestion
- The first parameter (timeout duration) for each of the following ``SocketSettings`` builder
  methods is updated from type ``int`` to ``long``:
```

**Review by rachel-mack — APPROVED** [review]
A question and a suggestion

File: source/upgrade.txt (hunk 1 above)
Comment (rustagir) [reply source/upgrade.txt:68]: yes I added a note in the PR description - I think this step wasnt done when you did the ticket, so just handled it here!

**Pull request merged**